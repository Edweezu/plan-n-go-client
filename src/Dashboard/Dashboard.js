import React from 'react'
import TripsContext from '../TripsContext'
import { Link } from 'react-router-dom'
// import Trip from '../Trip/Trip'
import TripApiService from '../services/trip-api-service'
import TripItem from '../TripItem/TripItem'

export default class Dashboard extends React.Component {

    static contextType = TripsContext

    state = {
        error: null,
    }
    
    componentDidMount() {
        TripApiService.getTrips()
            .then(trips => this.context.setTripList(trips))
            .catch(err => this.setError(err))
    }

    

    setError = (error) => {
        this.setState({
          error
        })
    }

    renderTrips() {
        const { tripList } = this.context
        console.log('triplisttt', tripList)
        return tripList.map(trip => 
            <li key={trip.id}>
                {/* <Link to={"/trip/" + trip.id}>{trip.trip_name}</Link>
                {' '}

                <button>Delete</button> */}
                <TripItem
                    id={trip.id}
                    name={trip.trip_name}
                    city={trip.city}
                    startDate={trip.start_date}
                    endDate={trip.end_date}
                    notes={trip.notes}
                />
            </li>       
        )
    }




    render () {
        const { error } = this.state
        return (
            <main className='dashboard'>
                <section className='upcoming-dashboard'>
                    <h2>Upcoming Trips</h2>
                    <ul>
                       {error ? <p className='error'>There was an error, try again</p>
                       : this.renderTrips()}
                    </ul>
                    <Link to='/add-trip'>Add a Trip</Link>
                </section>
            </main>
        )
    }
}