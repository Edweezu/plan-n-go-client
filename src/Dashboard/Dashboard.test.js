import React from 'react'
import toJson from 'enzyme-to-json'
import { shallow } from 'enzyme'
import Dashboard from './Dashboard'


describe('Dashboard component', () => {
    it('renders the given component', () => {
        const wrapper = shallow(<Dashboard />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })
})