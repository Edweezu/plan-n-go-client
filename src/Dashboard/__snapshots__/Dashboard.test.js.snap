// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard component renders the given component 1`] = `
<main
  className="dashboard"
>
  <section
    className="trips-dashboard-container"
  >
    <h1
      className="dashboard-header"
    >
      Dashboard
      <i
        className="fas fa-plus-circle"
        onClick={[Function]}
      />
    </h1>
    <ul />
  </section>
</main>
`;
